### ISSUE TITLE
Ledger - Sign Message hangs

### ISSUE THREAD

**Issue by GiovaniGuizzoBA** [secretkeylabs/sats-connect#106]
Using either `request("signMessage", { ... })` or `adapter.request("signMessage", { ... })` hangs with a ledger wallet connected. The promise is never resolved, even after confirming the signature on the ledger device.

**Comment by AymericNoel** [comment]
Hello, same happening here. 
Impossible to retrieve ledger signature....

**Comment by teebszet** [comment]
hi @GiovaniGuizzoYUB @AymericNoel , are you still experiencing the issue with the latest versions of sats-connect and Xverse chrome extension?

If so, do you have any more details on steps to reproduce this issue, or errors encountered?

thank you! 🙏

**Comment by GiovaniGuizzoBA** [comment]
I have dropped sats-connect and wrote my own code, actually. Sorry I cannot help you with that anymore.

**Comment by AymericNoel** [comment]
Hi @teebszet, Still experiencing this error with last version of Xverse chrome extension (0.38.1) and Sats-connect (2.5.0).

Steps to reproduce: 
- install sats-connect with npm
- install Xverse extension
- Import ledger account on Xverse (bc1 address)
- Connect wallet to website with : 
```js
 const providerResponse = await Wallet.request('getAccounts', {
          purposes: ['payment'],
          message: 'Address for signing in Ownership Tool'
        });
```
- Sign with : 
```js
   const response = await Wallet.request("signMessage", {
              address: this.walletAddress,
              message: this.generatedPhrase,
            })
            signature = response.result.signature
```
We can see the request on the ledger device and sign but we never get response from this call. But this is working with not ledger accounts.

**Comment by m-aboelenein** [comment]
> Hi @teebszet, Still experiencing this error with last version of Xverse chrome extension (0.38.1) and Sats-connect (2.5.0).
> 
> Steps to reproduce:
> 
> * install sats-connect with npm
> * install Xverse extension
> * Import ledger account on Xverse (bc1 address)
> * Connect wallet to website with :
> 
> ```js
>  const providerResponse = await Wallet.request('getAccounts', {
>           purposes: ['payment'],
>           message: 'Address for signing in Ownership Tool'
>         });
> ```
> 
> * Sign with :
> 
> ```js
>    const response = await Wallet.request("signMessage", {
>               address: this.walletAddress,
>               message: this.generatedPhrase,
>             })
>             signature = response.result.signature
> ```
> 
> We can see the request on the ledger device and sign but we never get response from this call. But this is working with not ledger accounts.

Hey @AymericNoel, we identified an issue specific to ledger on our end in the message signing request, the fix is already in the works and we are aiming to ship it soon, thanks for flagging this 🙏

**Comment by teebszet** [comment]
hi @AymericNoel this should be fixed from Xverse extension v0.39.0. thanks for reporting! 🙏
